ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/VertxPeerDiscoveryAgent.java

```diff
@@ -148,26 +148,31 @@ protected void handleListenerSetup(
     socket.exceptionHandler(this::handleException);
     socket.handler(this::handlePacket);
 
-    InetSocketAddress address =
+    final InetSocketAddress address =
         new InetSocketAddress(socket.localAddress().host(), socket.localAddress().port());
     addressFuture.complete(address);
   }
 
   @Override
   protected CompletableFuture<Void> sendOutgoingPacket(
       final DiscoveryPeer peer, final Packet packet) {
-    CompletableFuture<Void> result = new CompletableFuture<>();
-    socket.send(
-        packet.encode(),
-        peer.getEndpoint().getUdpPort(),
-        peer.getEnodeURL().getIpAsString(),
-        ar -> {
-          if (ar.failed()) {
-            result.completeExceptionally(ar.cause());
-          } else {
-            result.complete(null);
-          }
-        });
+    final CompletableFuture<Void> result = new CompletableFuture<>();
+    if (socket == null) {
+      result.completeExceptionally(
+          new RuntimeException("Discovery socket already closed, because Besu is closing down"));
```

Comment:
Is a RuntimeException appropriate here or something more specific?

Reply:
This should only happen when we are closing down (we have called stop in the VertxPeerDiscoveryAgent, which sets the socket to null), but there is still a message in the system that causes  sendOutgoingMessage() to be called.